Removes extend

cdmo opened the pull request:
I may be off, so please consider this more of a question. If you are `extend`ing the `MyMacros` class then you wouldn't need to namespace it in the call, it'd be just `accumulator << some_utility(record)`. If the code is meant to be called like `accumulator << MyUtility.some_utility(record)` as it is written in the example then `extend` is not required, all you'd need to do that is that `require 'my_utility'`. Right?

jrochkind commented:
Aha, I see what you mean.

EITHER you have a module mix-in that you "extend" and then call without a "namespace", OR you don't need `extend`, but will call a static/class method like `MyUtility.method`. 

The example mixed these two up, you're right. Nice catch, thanks! 

But I'd actually rather demo the first one, with "extend", because I think it's less obvious -- and a desired use case for providing nice custom "DSL" for local non-programmers to call. 

So I'd rather leave the "extned" in the docs, but fix the call site. Intersted in PR'ing that one instead? Feel free to suggest other language or code necessary to make it more clear whats going on in your opinion.

cdmo commented:
Does this version make more sense? I think it does to me. Sorry for the whitespace removal mucking up the diff.

jrochkind commented:
Is it basically just adding " and method called `my_some_macro`."?  

Oh, I see, `MyUtility.some_utility` was meant to be a different example!

Yeah, if this helps you, fine with me. i wonder if it would make it even more clear to actually show example after "and method called my_some_macro."? 

```ruby
module MyMacros
  def my_some_macro
 # ...
```

In the middle of writing that, I realized I forgot what the implementation of a macro looks like, all the more reason to maybe include a simple example there? Oh right, it returns a lambda.  I dunno, any thoughts?

cdmo commented:
I added a bit more in the way of examples and fixed some typos I noticed and one broken link too (hope that makes sense for a link to more info on macros).

jrochkind commented on doc/extending.md:
```diff
@@ -41,24 +41,43 @@ directory kept next to your traject config files:
+module MyUtility
+  def some_utility(record)
+    'some text' unless record.fields('505').empty?
+  end
```
For this to work as in the examples, I think it has to be `def self.some_utility`, no?  So it can be called like `MyUtility.some_utility`?

jrochkind commented:
Left one more comment, sorry it took me a while to circle back, sorry for the continued change requests, thanks for contributing to make the docs better!
